torch_em/transform/generic.py

```diff
@@ -105,6 +106,52 @@ def __call__(self, inputs):
         return inputs
 
 
+class ResizeLongestSideInputs:
+    def __init__(self, target_shape, target_length=512, is_label=False, is_rgb=False):
+        self.target_shape = target_shape
+        self.target_length = target_length
+        self.is_label = is_label
+        self.is_rgb = is_rgb
+
+    def _get_preprocess_shape(self, oldh, oldw):
+        """Inspired from Segment Anything."""
+        scale = self.target_length * 1.0 / max(oldh, oldw)
```

Comment:
Can't this just be derived from `target_shape`?

Reply:
True. On it!